src/DB/QueryBuilder.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DB;
+
+class QueryBuilder
+{
+    private ?QueryType $type;
+
+    private QueryInfo $info;
+
+    public function __construct(
+        private readonly QueryWriterFactory $factory,
+    ) {
+        $this->type = null;
+        $this->info = new QueryInfo();
+    }
+
+    public function select(string $table, array $fields = ['*']): QueryBuilder
+    {
+        $this->type = QueryType::SELECT;
```

Comment:
I would but space after this line to visually separate different type of logic. 

__optional__ but recommended, if doing here then change everywhere in class